os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -0,0 +1,688 @@
+	struct mtd_dev_s mtd;		/* MTD interface */
+	FAR struct spi_dev_s *dev;	/* Saved SPI interface instance */
+	uint8_t sectorshift;		/* 16 or 18 */
+	uint8_t pageshift;			/* 8 */
+	uint16_t nsectors;			/* 128 or 64 */
+	uint32_t npages;			/* 32,768 or 65,536 */
+#ifdef CONFIG_IMXRT_SUBSECTOR_ERASE
+	uint8_t subsectorshift;		/* 0, 12 or 13 (4K or 8K) */
+#endif
+};
+
+/************************************************************************************
+ * Private Function Prototypes
+ ************************************************************************************/
+
+/* Helpers */
+
+static void imxrt_lock(FAR struct spi_dev_s *dev);
+static inline void imxrt_unlock(FAR struct spi_dev_s *dev);
+static inline int imxrt_readid(struct imxrt_dev_s *priv);
+static void imxrt_waitwritecomplete(struct imxrt_dev_s *priv);
+static inline void imxrt_sectorerase(struct imxrt_dev_s *priv, off_t offset);
+static inline int imxrt_bulkerase(struct imxrt_dev_s *priv);
+static inline void imxrt_pagewrite(struct imxrt_dev_s *priv, FAR const uint8_t *buffer, off_t offset);
+
+/* MTD driver methods */
+
+static int imxrt_erase(FAR struct mtd_dev_s *dev, off_t startblock, size_t nblocks);
+static ssize_t imxrt_bread(FAR struct mtd_dev_s *dev, off_t startblock, size_t nblocks, FAR uint8_t *buf);
+static ssize_t imxrt_bwrite(FAR struct mtd_dev_s *dev, off_t startblock, size_t nblocks, FAR const uint8_t *buf);
+static ssize_t imxrt_read(FAR struct mtd_dev_s *dev, off_t offset, size_t nbytes, FAR uint8_t *buffer);
+#ifdef CONFIG_MTD_BYTE_WRITE
+static ssize_t imxrt_write(FAR struct mtd_dev_s *dev, off_t offset, size_t nbytes, FAR const uint8_t *buffer);
+#endif
+static int imxrt_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg);
+
+/************************************************************************************
+ * Private Data
+ ************************************************************************************/
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+
+
+
+
+
+/************************************************************************************
+ * Name: imxrt_lock
+ ************************************************************************************/
+
+static void imxrt_lock(FAR struct spi_dev_s *dev)
+{
+}
+
+/************************************************************************************
+ * Name: imxrt_unlock
+ ************************************************************************************/
+
+static inline void imxrt_unlock(FAR struct spi_dev_s *dev)
+{
+}
+
+/************************************************************************************
+ * Name: imxrt_readid
+ ************************************************************************************/
+
+static inline int imxrt_readid(struct imxrt_dev_s *priv)
+{
+	status_t status;
+
+	fvdbg("priv: %p\n", priv);
+	
```

Comment:
Please fix the indentation in following lines.

Reply:
fixed it